frontend/src/lib/uPlotLib/utils/getSeriesData.ts

```diff
@@ -23,20 +26,31 @@ const paths = (
 	return renderer(u, seriesIdx, idx0, idx1, extendGap, buildClip);
 };
 
-const getSeries = (
-	apiResponse?: MetricRangePayloadProps,
-	widgetMetaData: QueryData[] = [],
-	graphsVisibilityStates?: boolean[],
-): uPlot.Options['series'] => {
+const getSeries = ({
+	apiResponse,
+	widgetMetaData,
+	graphsVisibilityStates,
+	panelType,
+}: GetSeriesProps): uPlot.Options['series'] => {
 	const configurations: uPlot.Series[] = [
 		{ label: 'Timestamp', stroke: 'purple' },
 	];
 
 	const seriesList = apiResponse?.data.result || [];
 
+	const sortedSeriesList = seriesList.sort((a, b) => {
```

Comment:
@Rajat-Dabade: Does this guarantee that our data points map to the correct time stamps?

Reply:
Updated the logic, now in case of bar chart we are sorting the API response itself and then passing it to Uplot.